pkg/cmd/pulumi/convert.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/pulumi/pulumi/pkg/v3/codegen/dotnet"
+	gogen "github.com/pulumi/pulumi/pkg/v3/codegen/go"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/hcl2/syntax"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/nodejs"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/pcl"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/python"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/result"
+)
+
+func newConvertCmd() *cobra.Command {
+	var language string
+
+	cmd := &cobra.Command{
+		Use:    "convert",
+		Args:   cmdutil.MaximumNArgs(0),
+		Hidden: !hasExperimentalCommands(),
+		Short:  "Convert resource declarations into a pulumi program",
+		Long: "Convert resource declarations into a pulumi program.\n" +
+			"\n" +
+			"The PCL program to convert should be supplied on stdin.\n",
+		Run: cmdutil.RunResultFunc(func(cmd *cobra.Command, args []string) result.Result {
+
+			var programGenerator programGeneratorFunc
+			switch language {
+			case langDotnet:
+				programGenerator = dotnet.GenerateProgram
+			case langGo:
+				programGenerator = gogen.GenerateProgram
+			case langNodejs:
+				programGenerator = nodejs.GenerateProgram
+			case langPython:
+				programGenerator = python.GenerateProgram
+			default:
+				return result.Errorf("cannot generate programs for %v", language)
+			}
+
+			parser := syntax.NewParser()
+			err := parser.ParseFile(os.Stdin, "main.pp")
```

Comment:
This will be used in error messages, so I don't think we should reference a file that doesn't exist. 
```suggestion
                        err := parser.ParseFile(os.Stdin, "<stdin>")
```